frame/support/src/traits.rs

```diff
@@ -602,6 +602,29 @@ pub trait LockableCurrency<AccountId>: Currency<AccountId> {
 	);
 }
 
+/// A currency whose accounts can have balances which vest over time.
+pub trait VestingCurrency<AccountId>: Currency<AccountId> {
+	/// The quantity used to denote time; usually just a `BlockNumber`.
+	type Moment;
+
+	/// Get the amount that is currently being vested and cannot be transferred out of this account.
+	fn vesting_balance(who: &AccountId) -> Self::Balance;
+
+	/// Adds a vesting schedule to a given account.
+	///
+	/// If there already exists a vesting schedule for the given account, an `Err` is returned
+	/// and nothing is updated.
+	fn add_vesting_schedule(
+		who: &AccountId,
+		locked: Self::Balance,
+		per_block: Self::Balance,
+		starting_block: Self::Moment,
```

Comment:
Since the `Self::Moment` can be anything, why don't we change the `per_block` and `starting_block` to `per_moment` and `starting_moment`?
Also, would it work with timestamp pallet's UNIX milliseconds instead of block numbers? Seem like it should.